text/0000-best-practices-and-guidelines.md

```diff
@@ -0,0 +1,109 @@
+
+## Better consistency and user experience for users and authors
+
+Examples - 
+
+- A consistent naming scheme for `buildpacks`, `stack` IDs
+- A consistent way to "configure" `buildpacks` like specifying environment variables through a project descriptor.
+- A consistent format for `Bill-of-materials` metadata section to allow consistent rendering and parsing and easy integration with third-party dependency scanning tools.
+
+## Transfer of knowledge and best pratices accrued over time by experienced users to newer users
+
+Examples - 
+
+- Best practices around layer cache invalidation
+- Best practices for a `platform` operator
+
+
+## Better portability and reusability amongst different CNB ecosystems
+
+Examples - 
+
+- Using `platform` agnostic constructs when writing and distributing `buildpacks`
+
+
+## Allowing users to write maintainable and efficient `buildpacks` and `platforms`
+
+Examples - 
+
+- Creation of buildpacks that attempt to do one thing and only one thing well and composing them using `meta-buildpacks` (Paketo ideology)
+
+## Avoiding common pitfalls that new users may be unaware of
+
+Examples - 
+
+- `Bill-of-materials` needs to be regenerated on each run
+
+
+# What it is
+[what-it-is]: #what-it-is
+
+The `guidelines` would be contained in a sub-folder of the `buildpacks/docs` repository and it will be maintained by the `Learning team`. The folder should have separate sections that target different personas and list guidelines and best practices specific to them -
+
+- Buildpack users
+- Buildpack authors
+- Platform operators
+- Platform implementors
+
+These could be detailed sections or checklists that they can follow to ensure that they are following the best practices set by the guidelines.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+## Why should we *not* do this?
+
+More maintenance and support burden on the Learning team.
+
+# Alternatives
+[alternatives]: #alternatives
+
+## Create a new repository
+
+The guidelines and best-practices could be a separate repository under the buildpacks organization. This would allow the guidelines and best-practices to have a different review and approval process for the `guidelines`.
+
+## Extend the spec
+
+Have an extension to the spec that contains a list of `SHOULD`s and `MAY`s that capture the guidelines. This may however confuse users and may tie up something like the `spec` which is not geared towards end-users to `guidelines` which are supposed to be.
+
+
+# Prior Art
+[prior-art]: #prior-art
+
+- [Best coding practices](https://en.wikipedia.org/wiki/Best_coding_practices)
+- [Rust API guidelines](https://rust-lang.github.io/api-guidelines/about.html)
+- [PEP8 - Python Style-guide](https://www.python.org/dev/peps/pep-0008/)
+
+# Unresolved Questions
+[unresolved-questions]: #unresolved-questions
+
+- The exact set of guidelines that should be in the first cut of the repository
+- Topics and details to cover
```

Comment:
I wonder if creating a [discussion](https://github.com/buildpacks/community/discussions) would be a good way of seeding the initial set of guidelines.